spec/requests/api/v1/targets/index_spec.rb

```diff
@@ -3,28 +3,37 @@
 describe 'GET api/v1/targets/', type: :request do
   describe 'GET index' do
     let(:user) { create(:user) }
-    let!(:target) { create(:target) }
+    let!(:target) do
+      target = create(:target)
+      user.targets << target
+      return target
```

Comment:
no need of using return, anyway you shouldn't be doing this here. You can move this above the let(:user) and in the let(:user) do something like this: 
`let(:user) { create(:user, targets: [target] }`